ngi_visualizations/fpkm_scatter/fpkm_scatter.py

```diff
@@ -213,13 +255,98 @@ def plot_fpkm_scatter (sample_1, sample_2, x_lab, y_lab, output_fn=False, linear
     logging.info("Saving to {} and {}".format(png_fn, pdf_fn))
     plt.savefig(png_fn)
     plt.savefig(pdf_fn)
+    
+    # Return the filenames and R2
+    return {'png': png_fn, 'pdf': pdf_fn, output_fn:r_squared }
+
 
-    # Return the filenames
-    return {'png': png_fn, 'pdf': pdf_fn}
+def make_heatmap(r_values):
+    """
+    Takes a dict of r2 values and generates a heatmap using matplotlib
+    """
+    
+    data=r_values
+    #list the sample names
+    names=data.keys()
+    names=set()
+    for i in data.keys():
+        names.update(i.split('-'))
+    names=sorted(list(names))
+    
+    #If the sample looks like it's form a NGI project, do specific cleaning
+    ngi = re.compile(r'^P\d+_\d+')
+    clean_names=[]
+    for i in names:
+        i=i.split(".")[0]   #Remove everything after the first dot
+        j=re.search(ngi, i)
+        if j:
+            if i.endswith('Aligned'):
+                  i = i[:-7]
+        clean_names.append(i)
+    
+    matrix = []
+    #Generate the data matrix (list of lists) 
+    for idx, x in enumerate(names):
```

Comment:
A lot of this feels quite repetitive from above. Could we not do it once and pass around the matrix for both the output file and the heatmap?